projects/addon-doc/src/utils/coerce-value.ts

```diff
@@ -0,0 +1,25 @@
+export function coerceValue<T>(value?: T): T | number | string | boolean | null {
+    const prepared = String(value).trim();
+
+    if (isEmptyParamValue(prepared)) {
+        return null;
+    } else if (isBooleanParamValue(prepared)) {
+        return String(prepared) === 'true';
+    } else if (isNumberParamValue(prepared)) {
+        return Number(prepared);
+    }
+
+    return decodeURIComponent(prepared);
+}
+
+function isEmptyParamValue(value: string): boolean {
+    return ['undefined', 'null', 'NaN', 'Infinity'].includes(value);
+}
+
+function isBooleanParamValue(value: string): boolean {
+    return String(value) === 'true' || String(value) === 'false';
```

Comment:
should't we remove String() here? value is already string

Reply:
fixed